### ISSUE TITLE
Completion of subcommand options after last argument does not work

### ISSUE THREAD

**Issue by zx8** [pallets/click#919]
* Install latest `HEAD` version of `click`:
```
$ pip3 install -e git+git://github.com/pallets/click.git@132d66ac7d69a2b0e8f218a4cd39e50be3e0bcb9#egg=click
```
* Save the following in a file called `foo` on your `PATH`, ensuring it's executable:
```python
#!/usr/bin/env python3

import click

def _complete(ctx, args, incomplete):
    return ['abc', 'def', 'ghi', 'jkl', 'mno', 'pqr', 'stu', 'vwx', 'yz']

@click.group()
def entrypoint():
    pass

@click.command()
@click.option('--opt', autocompletion=_complete)
@click.argument('arg', nargs=-1)
def subcommand(opt, arg):
    pass

entrypoint.add_command(subcommand)

entrypoint()
```
* Source completion code
```sh
source <(_FOO_COMPLETE=source foo)
```
---
**Expected Behaviour**
```
$ foo subcommand --opt <TAB>
abc  def  ghi  jkl  mno  pqr  stu  vwx  yz

$ foo subcommand whatever --opt <TAB>
abc  def  ghi  jkl  mno  pqr  stu  vwx  yz
```
**Actual Behaviour**
```
# WORKS
$ foo subcommand --opt <TAB>
abc  def  ghi  jkl  mno  pqr  stu  vwx  yz

# BROKEN
$ foo subcommand whatever --opt <TAB>
<list of files in current directory (i.e. fallback when no results)>
```

**Comment by zx8** [comment]
Ping @untitaker

**Comment by atherra** [comment]
I can reproduce the issue. It appear to happen in _bashcomplete.resolve_ctx, which returns a None context in these circumstances, forcing get_choices to return early and empty-handed.

@stopthatcow may be interested in this.

**Comment by stopthatcow** [comment]
I will take a look this evening.

**Comment by stopthatcow** [comment]
Hi @zx8 Please try out my CL and be sure it fixes your issue. #931

**Comment by zx8** [comment]
@stopthatcow #931 doesn't fix this issue for me.

**Comment by stopthatcow** [comment]
Ok @zx8 try now. I had developed a test case that I thought was representative of your use case but it was not verbatim, now it is. I want to add a few more test cases but I think this fixes the issue.

**Comment by zx8** [comment]
Your new commit seems to fix the issue, thanks!

**Comment by stopthatcow** [comment]
@zx8  thats great, I rolled up the fix for this and #925 into PR #930 as it was a bit of an intertwined refactor. Feel free to take a look.
